crates/bevy_ecs/src/schedule/stepping.rs

```diff
@@ -0,0 +1,1305 @@
+        match system {
+            SystemIdentifier::Node(node_id) => {
+                self.behaviors.remove(&node_id);
+            }
+            SystemIdentifier::Type(type_id) => {
+                self.updates.insert(type_id, None);
+            }
+        }
+    }
+
+    // clear all system behaviors
+    fn clear_behaviors(&mut self) {
+        self.behaviors.clear();
+        self.first = None;
+    }
+
+    // apply system behavior updates by looking up the node id of the system in
+    // the schedule, and updating `systems`
+    fn apply_updates(&mut self, schedule: &Schedule) {
+        for (node_id, system) in schedule.systems().unwrap() {
+            let behavior = self.updates.get(&system.type_id());
+            match behavior {
+                None => continue,
+                Some(None) => {
+                    self.behaviors.remove(&node_id);
+                }
+                Some(Some(behavior)) => {
+                    self.behaviors.insert(node_id, *behavior);
+                }
+            }
+        }
+        self.updates.clear();
+
+        println!("apply_updates(): {:?}", self.behaviors);
+    }
+
+    fn skipped_systems(
+        &mut self,
+        schedule: &Schedule,
+        start: usize,
+        mut action: Action,
+    ) -> (FixedBitSet, Option<usize>) {
+        use std::cmp::Ordering;
+
+        // if our NodeId list hasn't been populated, copy it over from the
+        // schedule
+        if self.node_ids.len() != schedule.systems_len() {
+            self.node_ids = schedule.executable().system_ids.clone();
+        }
+
+        // Now that we have the schedule, apply any pending system behavior
+        // updates.  The schedule is required to map from system `TypeId` to
+        // `NodeId`.
+        if !self.updates.is_empty() {
+            self.apply_updates(schedule);
+        }
+
+        // if we don't have a first system set, set it now
+        if self.first.is_none() {
+            for (i, (node_id, _)) in schedule.systems().unwrap().enumerate() {
+                match self.behaviors.get(&node_id) {
+                    Some(SystemBehavior::AlwaysRun) | Some(SystemBehavior::NeverRun) => continue,
+                    Some(_) | None => {
+                        self.first = Some(i);
+                        break;
+                    }
+                }
+            }
+        }
+
+        let mut skip = FixedBitSet::with_capacity(schedule.systems_len());
+        let mut pos = start;
+
+        for (i, (node_id, system)) in schedule.systems().unwrap().enumerate() {
+            let behavior = self
+                .behaviors
+                .get(&node_id)
+                .unwrap_or(&SystemBehavior::Continue);
+            println!(
+                "skipped_systems(): systems[{}], pos {}, Action::{:?}, Behavior::{:?}, {}",
+                i,
+                pos,
+                action,
+                behavior,
+                system.name()
+            );
```

Comment:
See if you can get this message to only show up during testing, otherwise remove debug message